flytekit/common/local_workflow.py

```diff
@@ -0,0 +1,371 @@
+
+        return super().__call__(*args, **compiled_inputs)
+
+    @classmethod
+    def construct_from_class_definition(
+        cls,
+        inputs: List[_promise.Input],
+        outputs: List[Output],
+        nodes: List[_nodes.SdkNode],
+        metadata: _workflow_models.WorkflowMetadata = None,
+        metadata_defaults: _workflow_models.WorkflowMetadataDefaults = None,
+        disable_default_launch_plan: bool = False,
+    ) -> "PythonWorkflow":
+        """
+        This constructor is here to provide backwards-compatibility for class-defined Workflows
+
+        :param list[flytekit.common.promise.Input] inputs:
+        :param list[Output] outputs:
+        :param list[flytekit.common.nodes.SdkNode] nodes:
+        :param WorkflowMetadata metadata: This contains information on how to run the workflow.
+        :param flytekit.models.core.workflow.WorkflowMetadataDefaults metadata_defaults: Defaults to be passed
+            to nodes contained within workflow.
+        :param bool disable_default_launch_plan: Determines whether to create a default launch plan for the workflow or not.
+
+        :rtype: PythonWorkflow
+        """
+        for n in nodes:
+            for upstream in n.upstream_nodes:
+                if upstream.id is None:
+                    raise _user_exceptions.FlyteAssertion(
+                        "Some nodes contained in the workflow were not found in the workflow description.  Please "
+                        "ensure all nodes are either assigned to attributes within the class or an element in a "
+                        "list, dict, or tuple which is stored as an attribute in the class."
+                    )
+
+        id = _identifier.Identifier(
+            _identifier_model.ResourceType.WORKFLOW,
+            _internal_config.PROJECT.get(),
+            _internal_config.DOMAIN.get(),
+            _uuid.uuid4().hex,
+            _internal_config.VERSION.get(),
+        )
+        interface = _interface.TypedInterface({v.name: v.var for v in inputs}, {v.name: v.var for v in outputs})
+
+        output_bindings = [_literal_models.Binding(v.name, v.binding_data) for v in outputs]
+
+        return cls(
+            inputs=inputs,
+            nodes=nodes,
+            interface=interface,
+            output_bindings=output_bindings,
+            id=id,
+            metadata=metadata,
+            metadata_defaults=metadata_defaults,
+            disable_default_launch_plan=disable_default_launch_plan,
+        )
+
+    @property
+    def id(self):
+        return self._id
+
+    @id.setter
+    def id(self, new_id):
+        self._id = new_id
+
+    @property
+    def user_inputs(self) -> List[_promise.Input]:
+        """
+        :rtype: list[flytekit.common.promise.Input]
+        """
+        return self._user_inputs
+
+    def create_launch_plan(
+        self,
+        default_inputs: Dict[str, _promise.Input] = None,
+        fixed_inputs: Dict[str, Any] = None,
+        schedule=None,
```

Comment:
can we introduce type hints here?

Reply:
added it here, but it's missing from so many places... let's address in future PRs.